<?xml version="1.0"?>
<robot name="thymioid" xmlns:xacro="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface">

  <xacro:include filename="$(find thymio_driver)/models/base.urdf.xacro"/>

  <xacro:thymio />

  <link name="trunk_link">
    <visual>
      <geometry>
	<box size="0.03 0.11 0.06" />      
      </geometry>
      <material name="odroid">
	<color rgba="0 0.3 0.2 0.4"/>
      </material>      
    </visual>
  </link>

  <link name="camera_support_link">
    <visual>
      <geometry>
	<box size="0.022 0.03 0.02" />      
      </geometry>
      <material name="Black">
	<color rgba="0 0.0 0.0 0.4"/>
      </material>      
    </visual>
  </link>
  
  <link name="camera_body_link">
    <visual>
      <geometry>
	<box size="0.015 0.055 0.028" />      
      </geometry>
      <material name="Red">
	<color rgba="0.2 0.0 0.05 1.0"/>
      </material>      
    </visual>
  </link>

  <link name="camera_link">
    <visual>
      <geometry>
	<box size="0.016 0.02 0.01"/>
      </geometry>
      <material name="Grey">
	<color rgba="0.1 0.1 0.1 1.0"/>
      </material>      
    </visual>
  </link>
  
  <joint name="base_trunk_joint" type="fixed">
    <origin xyz="0.0 0.0 0.08" rpy="0 0 0"/>
    <parent link="base_link"/>
    <child link="trunk_link"/>
  </joint>

  <joint name="trunk_camera_joint" type="fixed">
    <origin xyz="0.025 0 0.03" rpy="0 0 0"/>
    <parent link="trunk_link"/>
    <child link="camera_support_link"/>
  </joint>
  
  <joint name="camera_body_support_joint" type="fixed">
    <origin xyz="0.015 0 0" rpy="0 0.2618 0"/>
    <parent link="camera_support_link"/>
    <child link="camera_body_link"/>
  </joint>

  <joint name="camera_frame_joint" type="fixed">
    <origin xyz="0.0 0 0" rpy="-1.5708 0 -1.5708"/>
    <parent link="camera_body_link"/>
    <child link="camera_link"/>
  </joint>
  
</robot>
